OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -46,6 +47,8 @@ public ProvidesPrerequisite(ActorInitializer init, ProvidesPrerequisiteInfo info
 			if (string.IsNullOrEmpty(prerequisite))
 				prerequisite = init.Self.Info.Name;
 
+			techTree = init.Self.Owner.PlayerActor.Trait<TechTree>();
```

Comment:
This is used at most once, so there is no benefit in querying and caching this here. Look it up when you need it in the capture callback.